src/realm/parser/keypath_mapping.cpp

```diff
@@ -67,6 +63,73 @@ util::Optional<std::string> KeyPathMapping::get_mapping(TableKey table_key, cons
     return ret;
 }
 
+bool KeyPathMapping::add_table_mapping(ConstTableRef table, std::string alias)
```

Comment:
Wow, that was quick :D. Thanks for adding this 💯 

Would it be possible to also add support for this to the ObjectStore schemas? Because right now we use `realm::populate_keypath_mapping(*mapping, *shared_realm)` as the primary way of creating the mappings.

On `Property` we have `name` and `public_name`, adding something similar to `ObjectSchema` seems like the quickest way do to this?

Reply:
Ok I've added this as well, you'll have to construct your ObjectSchemas with the alias to take advantage of `populate_keypath_mapping` doing this for you automatically.